dev-clean/src/main/scala/sai/llsc/states/GenericDefs.scala

```diff
@@ -80,6 +82,8 @@ trait Opaques { self: SAIOps with BasicDefs =>
     def noop: Rep[Value] = "llsc-external-wrapper".reflectWith[Value]("noop")
   }
 
+  def getString(ptr: Rep[Value], s: Rep[SS]): Rep[String] = "get-string".reflectWith[String](ptr, s)
```

Comment:
nit: you could just write `"get_string"` then you don't need to specify codegen for it.